File: maj_produit_caisse.php
<?php
header("Access-Control-Allow-Origin: *");
error_reporting(E_ALL);
$bdd=mysqli_connect('195.114.27.208','web','admin','aircotedivoire');
$query="SELECT pr_cd_pr,pr_desi,tr_prix,pr_famille FROM trolley,produit,produit_plus where tr_code=2511 and tr_cd_pr=produit.pr_cd_pr and produit_plus.pr_cd_pr=produit.pr_cd_pr";
echo $query;
$result = mysqli_query($bdd,$query);
$outp="";
while ($row = mysqli_fetch_array($result)){
	$pr_cd_pr=$row['pr_cd_pr'];
	$pr_desi=$row['pr_desi'];
	$pr_prx_vte=$row['tr_prix'];
	$pr_famille=$row['pr_famille'];
	if ($pr_famille=6 || $pr_famille=7){$famille="Bijouterie";}
	if ($pr_famille=21 || $pr_famille=19){$famille="Accessoires";}
	if ($pr_famille=1 ||$pr_famille=10){$famille="Parfums Femmes";}
	if ($pr_famille=3){$famille="Parfums Hommes";}
	if ($pr_famille=2 || $pr_famille=8 ){$famille="Coffrets Parfums";}
	if ($pr_famille=5){$famille="Comestiques";}
	if ($pr_famille=4){$famille="Montres";}
	
	$pr_remise=0;
	if ($outp == ""){$outp = '[{"pr_cd_pr":"'.$pr_cd_pr.'","pr_desi":"'.$pr_desi.'","pr_prx_vte":'.$pr_prx_vte.',"pr_remise":'.$pr_remise.',"pr_famille":'.$pr_famille.'}';}
	else{
		$outp .= ',{"pr_cd_pr":"'.$pr_cd_pr.'","pr_desi":"'.$pr_desi.'","pr_prx_vte":'.$pr_prx_vte.',"pr_remise":'.$pr_remise.',"pr_famille":'.$pr_famille.'}';	
	}
}
$outp.=']';
echo "$outp";
?> 

File: maj_produit_caisse.pl
#!/usr/bin/perl
use DBI();
use CGI();
use JSON;
#$html=new CGI;
	print $html->header(-type=>"text/plain", -Access_Control_Allow_Origin=>"*",
	-access_control_allow_headers => 'content-type,X-Requested-With',
-access_control_allow_methods => 'GET,POST,OPTIONS',
-access_control_allow_credentials => 'true',
	
	);
print "Content-type:application/json\n\n";
require "../oasix/../oasix/outils_perl2.pl";
require("./src/connect.src");

$trolley=2511;
$outp="";
&save("create temporary table ordre_temp (famille int(3),desi varchar(50),code int(10),code_court int(5))"); 
$query = "select tr_cd_pr from trolley where tr_code='$trolley'";
$sth=$dbh->prepare($query);
$sth->execute();
while (($code)=$sth->fetchrow_array){
	&famille($code);
	$desi_tpe=&get("select pr_desi from produit where pr_cd_pr='$code'");
	$desi_tpe=lc($desi_tpe);
	$desi_tpe=ucfirst($desi_tpe);
	$desi_tpe=~s/\&//g;
	&save ("insert into ordre_temp values ('$fa_cat','$desi_tpe','$code','$code_court')");
}
$query="select famille,desi,code,code_court from ordre_temp order by famille,desi";
$sth=$dbh->prepare($query);
$sth->execute();
while (($famille,$pr_desi,$code,$code_court)=$sth->fetchrow_array){
	$pr_prix=&get("select tr_prix/100 from trolley where tr_code='$trolley' and tr_cd_pr='$code'")+0; 
	$pr_prix=int($pr_prix);
	$pr_prix0=$pr_prix;
	$pr_prix1=&get("select tr_prix/100 from trolley where tr_code='$trolley2' and tr_cd_pr='$code'")+0;
	$pr_prix1=int($pr_prix1);
	if ($pr_prix1==0){
			$pr_prix1=int($pr_prix*$cours_xof/1000)*1000;
	}
	$pr_cd_pr=$code;
	$pr_prx_vte=$pr_prix0;
	$pr_remise=$pr_prix1;
    $pr_desi=~s/\"//;
	$pr_desi=~s/\'//;
	$pr_desi=~s/\://;
	$pr_desi=~s/\.//;
	$pr_desi=~s/\{//;
	$pr_desi=~s/\}//;
	$pr_desi=~s/\]//;
	$pr_desi=~s/\[//;
	$pr_desi=~s/\n//;
	if ($outp eq ""){$outp = '[{"pr_cd_pr":"'.$pr_cd_pr.'","pr_desi":"'.$pr_desi.'","pr_prx_vte":'.$pr_prx_vte.',"pr_remise":'.$pr_remise.'}';}
	else{
		$outp .= ',{"pr_cd_pr":"'.$pr_cd_pr.'","pr_desi":"'.$pr_desi.'","pr_prx_vte":'.$pr_prx_vte.',"pr_remise":'.$pr_remise.'}';	
	}
}
$outp.=']';
